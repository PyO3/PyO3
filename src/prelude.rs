//! PyO3's prelude.
//!
//! The purpose of this module is to alleviate imports of many commonly used items of the PyO3 crate
//! by adding a glob import to the top of pyo3 heavy modules:
//!
//! ```
//! # #![allow(unused_imports)]
//! use pyo3::prelude::*;
//! ```

pub use crate::conversion::{FromPyObject, IntoPyObject};
#[allow(deprecated)]
pub use crate::conversion::{IntoPy, ToPyObject};
pub use crate::err::{PyErr, PyResult};
pub use crate::instance::{Borrowed, Bound, Py, PyObject};
pub use crate::marker::Python;
pub use crate::pycell::{PyRef, PyRefMut};
pub use crate::pyclass_init::PyClassInitializer;
pub use crate::types::{PyAny, PyModule};

#[cfg(feature = "macros")]
pub use pyo3_macros::{
    pyclass, pyfunction, pymethods, pymodule, FromPyObject, IntoPyObject, IntoPyObjectRef,
};

#[cfg(feature = "macros")]
pub use crate::wrap_pyfunction;
#[cfg(feature = "macros")]
#[allow(deprecated)]
pub use crate::wrap_pyfunction_bound;

pub use crate::types::any::PyAnyMethods;
pub use crate::types::weakref::PyWeakrefMethods;
