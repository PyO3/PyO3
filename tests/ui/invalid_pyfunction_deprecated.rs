use pyo3::prelude::*;

#[pyfunction]
#[pyo3(deprecated)]
fn deprecated_function() {}

#[pyfunction]
#[pyo3(deprecated = )]
fn deprecated_function2() {}

#[pyfunction]
#[pyo3(deprecated = "first deprecated")]
#[pyo3(deprecated = "second deprecated")]
fn deprecated_function3() {}

fn main() {}
