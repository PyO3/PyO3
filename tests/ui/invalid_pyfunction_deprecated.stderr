error: expected `=`
 --> tests/ui/invalid_pyfunction_deprecated.rs:4:18
  |
4 | #[pyo3(deprecated)]
  |                  ^

error: unexpected end of input, expected string literal
 --> tests/ui/invalid_pyfunction_deprecated.rs:8:21
  |
8 | #[pyo3(deprecated = )]
  |                     ^

error: only one `deprecated` warning may be specified
  --> tests/ui/invalid_pyfunction_deprecated.rs:13:8
   |
13 | #[pyo3(deprecated = "second deprecated")]
   |        ^^^^^^^^^^
